helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java

```diff
@@ -69,7 +69,14 @@ public void testForceDeleteJobFromJobQueue() throws InterruptedException {
     Assert
         .assertNotNull(_driver.getJobContext(TaskUtil.getNamespacedJobName(jobQueueName, "job2")));
 
-    // The following force delete for the job should go through without getting an exception
+    // Force deletion can have Exception thrown as controller is writing to propertystore path too.
+    // https://github.com/apache/helix/issues/1406, also force deletion may not be safe.
+    // Thus, we stop pipeline to make sure there is not such race condition.
+    _gSetupTool.getClusterManagementTool().enableCluster(CLUSTER_NAME, false);
+    Thread.sleep(3000);
+    // note this sleep is critical as it would take time for controller to stop.
```

Comment:
Can you change to "TODO", and put before the sleep line?

Reply:
changed.